flink-core/src/main/java/org/apache/flink/api/common/io/DelimitedInputFormat.java

```diff
@@ -182,8 +186,37 @@ protected DelimitedInputFormat(Path filePath, Configuration configuration) {
 		}
 		loadConfigParameters(configuration);
 	}
-	
-	
+
+	/**
+	 * Get the character set used for the row delimiter, field delimiter,
+	 * comment string, and {@link FieldParser}.
+	 *
+	 * @return the charset
+	 */
+	@PublicEvolving
+	public Charset getCharset() {
+		if (this.charset == null) {
+			this.charset = Charset.forName(charsetName);
+		}
+		return this.charset;
+	}
+
+	/**
+	 * Set the name of the character set used for the row delimiter, field
+	 * delimiter, comment string, and {@link FieldParser}.
+	 *
+	 * The delimeters and comment string are interpreted when set. Each
```

Comment:
Can we change the encoding of the current delimiter using the old charset and the new charset?
That would also allow to set a new charset without the need to set `\n` again to have a newline delimiter.